src/services/jwp_entitlement.service.ts

```diff
@@ -0,0 +1,14 @@
+import InPlayer from '@inplayer-org/inplayer.js';
+import { injectable } from 'inversify';
+
+@injectable()
+export default class JWPEntitlementService {
+  getJWPMediaToken = async (configId: string = '', mediaId: string) => {
+    try {
+      const { data } = await InPlayer.Asset.getSignedMediaToken(configId, mediaId);
```

Comment:
I see that the JWP entitlement service is using the InPlayer library. Should this be part of the integration? E.g. mixing Cleeng with JWP Entitlement will never work, right?

Reply:
JWP entitlement service won't work with cleeng as clients will need a JW `clientId`.

We can potentially have several services here:

`EntitlementService` with one abstract `getMediaToken` method.
`JWPEntitlementService` with an implementation of `getMediaToken`.
`SelfHostedEntitlementService` with an implementation of `getMediaToken`.

Though they are quite different and work with different arguments. We can't be sure that in case there is JW integration then only JWPEntitlementService is to be used. You can take a look here, that the condition is quite complex: `src/hooks/useContentProtection.ts`.

Looks like we will also have to bind services conditionally based on the configuration. In case `signingService` is set  -> then bind `SelfHostedEntitlementService`, if not check integration and maybe bind `JWPEntitlementService`.

```
container.bind('CONFIGURATION').toDynamicValue((context) => {
  return context.container.get(AppController).getConfiguration();
});
```

I think I would prefer combining them back into one one service and refactoring them later, as I may break some logic there without full e2e coverage.

Reply:
Hmm, I like using the same method as used for integrations, but that sounds like a big change. 

I'm fine with keeping it "as is" for now and think about this later. @dbudzins what do you think?